### PR TITLE
add an env.subscription to the enviroment values

### PR DESCRIPTION
**Pull request by aabughosh**

### CODE REVIEW

File: pkg/autodiscover/autodiscover_operators.go [hunk 1]
```diff
@@ -19,11 +19,14 @@ package autodiscover
 import (
 	"context"
 
+	helmclient "github.com/mittwald/go-helm-client"
```
Comment (edcdavid) [inline pkg/autodiscover/autodiscover_operators.go:22]: Use the openshift helm client-go instead:
	clienthelm "github.com/openshift/client-go/helm/clientset/versioned/typed/helm/v1beta1"

File: pkg/autodiscover/autodiscover.go [hunk 2]
```diff
@@ -104,6 +115,12 @@ func DoAutoDiscover() DiscoveredTestData {
 	data.DebugPods = findPodsByLabel(oc.Coreclient, debugLabels, debugNS)
 	data.Crds = FindTestCrdNames(data.TestData.CrdFilters)
 	data.Csvs = findOperatorsByLabel(oc.OlmClient, []configuration.Label{{Name: tnfCsvTargetLabelName, Prefix: tnfLabelPrefix, Value: tnfCsvTargetLabelValue}}, data.TestData.TargetNameSpaces)
+	data.Subscriptions = findSubscriptions(oc.OlmClient, []configuration.Label{{Name: tnfCsvTargetLabelName, Prefix: tnfLabelPrefix, Value: tnfCsvTargetLabelValue}}, data.TestData.TargetNameSpaces)
+	data.HelmList = getHelmList(oc.RestConfig, data.TestData.TargetNameSpaces)
+	openshiftVersion, _ := getOpenshiftVersion(oc.OClient)
```
Comment (hamadise) [inline pkg/autodiscover/autodiscover.go:120]: check the returned error.

File: pkg/autodiscover/autodiscover.go [hunk 3]
```diff
@@ -104,6 +115,12 @@ func DoAutoDiscover() DiscoveredTestData {
...
+	data.OpenshiftVersion = openshiftVersion
+	k8sVersion, _ := oc.K8sClient.DiscoveryClient.ServerVersion()
```
Comment (hamadise) [inline pkg/autodiscover/autodiscover.go:122]: check the returned error

File: pkg/autodiscover/autodiscover_operators.go [hunk 4]
```diff
@@ -51,3 +54,57 @@ func findOperatorsByLabel(olmClient *clientOlm.Clientset, labels []configuration
...
+
+	return subscriptions
+}
+
+func getHelmList(restConfig *rest.Config, namespaces []configuration.Namespace) [][]*release.Release {
+	helmlist := [][]*release.Release{}
+	for _, ns := range namespaces {
+		opt := &helmclient.RestConfClientOptions{
+			Options: &helmclient.Options{
+				Namespace:        ns.Name, // Change this to the namespace you wish the client to operate in.
```
Comment (hamadise) [inline pkg/autodiscover/autodiscover_operators.go:88]: comment should be removed

File: pkg/autodiscover/autodiscover_operators.go [hunk 5]
```diff
@@ -51,3 +54,57 @@ func findOperatorsByLabel(olmClient *clientOlm.Clientset, labels []configuration
...
+
+	logrus.Infof("Found %d Subscriptions:", len(subscriptions))
+	for i := range subscriptions {
+		logrus.Infof(" Subscriptions name: %s (ns: %s)", subscriptions[i].Name, subscriptions[i].Namespace)
+	}
+
+	return subscriptions
+}
+
+func getHelmList(restConfig *rest.Config, namespaces []configuration.Namespace) [][]*release.Release {
```
Comment (hamadise) [inline pkg/autodiscover/autodiscover_operators.go:83]: use []namespaces directly.

File: pkg/autodiscover/autodiscover_operators.go (hunk 1 above)
Comment (edcdavid) [reply pkg/autodiscover/autodiscover_operators.go:22]: Disregard, this only list helm repositories.
deliedit@fedora newrepo]$ oc get helmchartrepositories -owide
NAME                    AGE
openshift-helm-charts   41d
[deliedit@fedora newrepo]$

File: pkg/autodiscover/autodiscover.go (hunk 2 above)
Comment (edcdavid) [reply pkg/autodiscover/autodiscover.go:120]: Should this be fatal? I would just print an error an put an empty string

**Review by edcdavid — APPROVED** [review]

File: pkg/autodiscover/autodiscover_operators.go [hunk 6]
```diff
@@ -51,3 +54,54 @@ func findOperatorsByLabel(olmClient *clientOlm.Clientset, labels []configuration
 
 	return csvs
 }
+func findSubscriptions(olmClient *clientOlm.Clientset, labels []configuration.Label, namespaces []configuration.Namespace) []olmv1Alpha.Subscription {
```
Comment (hamadise) [inline pkg/autodiscover/autodiscover_operators.go:57]: use []string instead of []configuration.Namespace

**Pull request merged**